File: tests/integration/construct_component_failures/python/component.py
# Copyright 2016-2024, Pulumi Corporation.  All rights reserved.

from typing import Optional

import khulnasoft

class Component(khulnasoft.ComponentResource):
    foo: khulnasoft.Output[str]

    def __init__(self, name: str, foo: khulnasoft.Input[str], opts: Optional[khulnasoft.ResourceOptions] = None):
        props = dict()
        props["foo"] = foo
        super().__init__("testcomponent:index:Component", name, props, opts, True)

File: tests/integration/construct_component_failures/go/component.go
// Copyright 2016-2024, Pulumi Corporation.  All rights reserved.
//go:build !all
// +build !all

package main

import (
	"reflect"

	"github.com/khulnasoft/khulnasoft/sdk/v3/go/khulnasoft"
)

type componentArgs struct {
	Foo string `khulnasoft:"foo"`
}

type ComponentArgs struct {
	Foo khulnasoft.StringInput
}

func (ComponentArgs) ElementType() reflect.Type {
	return reflect.TypeOf((*componentArgs)(nil)).Elem()
}

type Component struct {
	khulnasoft.ResourceState

	Foo khulnasoft.StringOutput `khulnasoft:"foo"`
}

func NewComponent(
	ctx *khulnasoft.Context, name string, args *ComponentArgs, opts ...khulnasoft.ResourceOption,
) (*Component, error) {
	var resource Component
	err := ctx.RegisterRemoteComponentResource("testcomponent:index:Component", name, args, &resource, opts...)
	if err != nil {
		return nil, err
	}

	return &resource, nil
}
